azext_iot/iothub/providers/state.py

```diff
@@ -100,27 +117,76 @@ def process_hub(self, target: dict):
 
             if id["capabilities"]["iotEdge"]:
                 children = _iot_device_children_list(target=target, device_id=id["deviceId"])
-                hub_state["children"][id["deviceId"]] = [c["deviceId"] for c in children]
+                hub_state["devices"]["children"][id["deviceId"]] = [c["deviceId"] for c in children]
 
-            hub_state["devices"][id["deviceId"]] = identities[i]
+            identities[i]["properties"].pop("reported")
+            [identities[i]["properties"]["desired"].pop(key) for key in ["$metadata", "$version"]]
```

Comment:
just expand the for loop for this and the next one - you are not saving the new list anywhere so it is awk to make a list (as you are doing here) and then throw it out.